### ISSUE TITLE
Position Icon Orientation does not follow direction of movement if route is active

### ISSUE THREAD

**Issue by daniel-callejas-sevilla** [osmandapp/OsmAnd#21428]
### Description

I expect the Position Icon arrow to point in the direction of movement, but when OsmAnd is set to follow a `.gpx` route, the Position Icon arrow gets confused and points towards the expected route, instead of the current position.

This is perhaps related to, but not the same as, #20858 and #21067. In my case the Position Icon is not stuck, it's just drunk. And the screen is on all the time.

Screen is on all the time, phone is charging.

### Steps to reproduce

* Define a random route by tapping on points on the map.
* Enable trip recording, so that my actual position appears drawn on the map.
* OsmAnd calculates the itinerary from my position to the start of route.
* Start driving somewhere else. OsmAnd keeps rerouting me to the start of route. During rerouting, Position Icon arrow has a tendency to point to the next turn in the itinerary, instead of my actual direction of movement.

### Actual result

Screen rotates correctly in the direction of movement, but Position Icon arrow points in a direction that does not match my movement. See video.

https://github.com/user-attachments/assets/413fd44e-e96a-4bdd-b842-64fe58ffed9c

### Expected result

Position Icon arrow should at all times point in the direction of movement, regardless of any route or redirection.

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version: 4.9.7
Android/iOS version: Android 15
Device model: Google Pixel 8a
Crash-Logs: no crash
```

**Comment by Norbert80** [comment]
I see the same issue. Let me know if attaching an additional screen recording helps.

**Comment by vshcherb** [comment]
// TODO this code is not synced with location (bearing should be only approximated if location is on the line)
https://github.com/osmandapp/Osmand/blob/dc048ebf39df30af0b956e0f67bb9786daef7d39/OsmAnd/src/net/osmand/plus/routing/RoutingHelperUtils.java#L139

**Comment by vshcherb** [comment]
https://github.com/osmandapp/OsmAnd/commit/133191e840a47f1e81c14fcaef78c0910832ed08

**Comment by Zirochkabila** [comment]
OsmAnd~ 5.0.0#3860m, Released: 2024-12-02

Still, reproduce
to reproduce it, you need to start navigation from the start point of the track - start navigation and start simulating the location along the track. At the moment of deviation from the actual route, you may encounter a bug


https://github.com/user-attachments/assets/b2855e23-5cae-441a-b896-7eaecd59810e

**Comment by vshcherb** [comment]
Based on the video provided it's expected. As there is a preference to follow by route (delay of animation), original bug is not reproduced

**Comment by daniel-callejas-sevilla** [comment]
The video provided by @Zirochkabila above reproduces the same issue that I posted originally: the arrow points in a direction that does not match movement.

**Comment by vshcherb** [comment]
It happens within a single animation rotation and relocation, in your video it was visible that location was moving with wrong bearing enabled

**Comment by daniel-callejas-sevilla** [comment]
I have trouble understanding the difference. In both videos I see the blue arrow point in a direction that does not match direction of movement.

My understanding is that the blue arrow makes a guess about what the next heading is going to be, using the the defined as a predictor. This behavior (guessing a heading from the route) should be disabled because it induces the user into error (e.g. when navigating an unfamiliar intersection, and having only a fraction of a second to compare the heading of the blue arrow compared to the surroundings of the vehicle to decide). 

In other words: Having the arrow 100% pegged to the detected direction of movement allows the user to make consistent decisions (user knows that the arrow is either accurate, or at least lags behind the actual orientation by a predictable period). Having the arrow alternate between route and direction of movement removes that predictability.
